### PR TITLE
Define workflow access perms and roles requestable

### PR DESCRIPTION
**Pull request by kimlisa**
redo of https://github.com/gravitational/webapps/pull/203

#### Description
- Remove enterprise features: workflow routes, config, access request session storage
- Set/Check AccessRequest access perms
- Set Roles Requestable for UserContext

### CODE REVIEW

**Review by alex-kovoy — APPROVED** [review]

File: packages/teleport/src/config.ts [hunk 1]
```diff
@@ -108,7 +111,8 @@ const cfg = {
     nodeTokenPath: '/v1/enterprise/nodes/token',
     nodeScriptPath: '/scripts/:token/install-node.sh',
     appNodeScriptPath: '/scripts/:token/install-app.sh?name=:name&uri=:uri',
-    requestAccessPath: '/v1/enterprise/accessrequest/:requestId?',
+    accessRequestPath: '/v1/enterprise/accessrequest/:requestId?',
```
Comment (alex-kovoy) [inline packages/teleport/src/config.ts:114]:
```
accessRequestPath: '/v1/enterprise/accessrequest/:requestId?',
accessRequestsPath: '/v1/enterprise/accessrequest?id=:requestId?',
```

we are not going to open source these, so lets move it to E config. The reason why some of these endpoints reside in OSS is becauase roles and users are scheduled to be open sourced soon. But not the workflows.

Comment (alex-kovoy) [reply packages/teleport/src/config.ts:114]:
```
accessRequestPath: '/v1/enterprise/accessrequest/:requestId?',
accessRequestsPath: '/v1/enterprise/accessrequest?id=:requestId?',
```

Why are there 2 ways to pass a request ID?

Comment (kimlisa) [reply packages/teleport/src/config.ts]: moved to enterprise and deleted additional path

File: packages/teleport/src/services/user/makeUserContext.ts [hunk 2]
```diff
@@ -26,13 +26,15 @@ export default function makeUserContext(json: any): UserContext {
...
   const accessStrategy = json.accessStrategy || defaultStrategy;
+  const rolesRequestable = json.rolesRequestable;
 
   return {
     username,
     authType,
     acl,
     cluster,
     accessStrategy,
+    rolesRequestable,
```
Comment (alex-kovoy) [inline packages/teleport/src/services/user/makeUserContext.ts:37]: requestableRoles is better

**Review by alex-kovoy — APPROVED** [review]
with comment

File: packages/teleport/src/services/clusters/makeCluster.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ export default function makeCluster(json): Cluster {
     lastConnected: new Date(lastConnected),
     connectedText,
     status,
-    url: cfg.getClusterRoute(clusterId),
+    url: clusterId ? cfg.getClusterRoute(clusterId) : '',
```
Comment (alex-kovoy) [inline packages/teleport/src/services/clusters/makeCluster.ts:47]: why?

File: packages/teleport/src/services/user/user.test.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * limitations under the License.
+ */
+
+import user from './user';
+import api from 'teleport/services/api';
+import { defaultAccess } from 'teleport/services/user/makeAcl';
+import { defaultStrategy } from 'teleport/services/user/makeUserContext';
+
+test('fetch user context, null response gives proper default values', async () => {
+  jest.spyOn(api, 'get').mockResolvedValue(null);
```
Comment (alex-kovoy) [inline packages/teleport/src/services/user/user.test.ts:23]: why?

**Review by alex-kovoy — APPROVED** [review]
with comments

File: packages/shared/libs/stores/useStore.ts [hunk 5]
```diff
@@ -27,7 +27,7 @@ export default function useStore<T extends Store<any>>(store: T): T {
     function syncState() {
       // do not re-render if state has not changed since last call
       if (memoizedState !== store.state) {
-        rerender({});
+        rerender({} as any);
```
Comment (alex-kovoy) [inline packages/shared/libs/stores/useStore.ts:30]: useState<any>()

Comment (kimlisa) [reply packages/shared/libs/stores/useStore.ts:30]: revised

File: packages/teleport/src/services/clusters/makeCluster.ts (hunk 3 above)
Comment (kimlisa) [reply packages/teleport/src/services/clusters/makeCluster.ts:47]: reverted

File: packages/teleport/src/services/user/user.test.ts (hunk 4 above)
Comment (kimlisa) [reply packages/teleport/src/services/user/user.test.ts:23]: revised

**Pull request merged**